integrations/cohere/tests/test_cohere_chat_generator.py

```diff
@@ -200,6 +256,36 @@ def __call__(self, chunk: StreamingChunk) -> None:
         assert "prompt_tokens" in message.meta["usage"]
         assert "completion_tokens" in message.meta["usage"]
 
+    @pytest.mark.skipif(
+        not os.environ.get("COHERE_API_KEY", None) and not os.environ.get("CO_API_KEY", None),
+        reason="Export an env var called COHERE_API_KEY/CO_API_KEY containing the Cohere API key to run this test.",
+    )
+    @pytest.mark.integration
+    @pytest.mark.asyncio
+    async def test_live_run_streaming_async(self):
+        class Callback:
+            def __init__(self):
+                self.responses = ""
+                self.counter = 0
+
+            async def __call__(self, chunk: StreamingChunk) -> None:
+                self.counter += 1
+                self.responses += chunk.content if chunk.content else ""
+
+        callback = Callback()
```

Comment:
please use something like this: https://github.com/deepset-ai/haystack-core-integrations/blob/b26aaec1700c5bab562226ce4df5e838d9dd294d/integrations/mistral/tests/test_mistral_chat_generator_async.py#L232